client/src/context/useBoardContext.tsx

```diff
@@ -0,0 +1,27 @@
+import { useState, useContext, createContext, FunctionComponent, useCallback } from 'react';
+import { Column } from '../interface/Column';
+import { columns } from '../mocks/mockBoardData';
+
+interface IBoardContext {
+  board: Column[];
+  updateBoard: (board: Column[]) => void;
+}
+
+export const BoardContext = createContext<IBoardContext>({
+  board: [],
+  updateBoard: () => null,
+});
+
+export const BoardProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [board, setBoard] = useState(columns);
+
+  const updateBoard = useCallback((newBoard: Column[]) => {
+    setBoard(newBoard);
+  }, []);
```

Comment:
Could you elaborate on your choice for useCallback here?

Reply:
Doesn't do much here except retain the reference to the function `updateBoard` every time it is called.